src/status_im/common/bottom_sheet/view.cljs

```diff
@@ -57,83 +56,77 @@
-         ;; backdrop
-         [rn/pressable
-          {:on-press #(rf/dispatch [:hide-bottom-sheet])
-           :style    {:flex 1}}
-          [reanimated/view
-           {:style (reanimated/apply-animations-to-style
-                    {:opacity bg-opacity}
-                    {:flex 1 :background-color colors/neutral-100-opa-70})}]]
-         ;; sheet
-         [gesture/gesture-detector {:gesture sheet-gesture}
-          [reanimated/view
-           {:style (reanimated/apply-animations-to-style
-                    {:transform [{:translateY translate-y}]}
-                    (style/sheet insets window-height selected-item))}
-           (when shell?
-             [blur/ios-view
-              {:style         style/shell-bg
-               :blur-radius   (or blur-radius 20)
-               :blur-type     :transparent
-               :overlay-color :transparent}])
-           (when selected-item
-             [rn/view
-              {:on-layout #(reset! item-height (.-nativeEvent.layout.height ^js %))
-               :style
-               (style/selected-item theme top bottom selected-item-smaller-than-sheet? border-radius)}
-              [selected-item]])
+(defn view
+  [{:keys [hide? insets]}
+   {:keys [content selected-item padding-bottom-override border-radius on-close shell?
+           gradient-cover? customization-color hide-handle? blur-radius]
+    :or   {border-radius 12}}]
+  (let [theme                             (quo.theme/use-theme-value)
+        sheet-height                      (rn/use-ref-atom 0)
+        item-height                       (rn/use-ref-atom 0)
+        {window-height :height}           (rn/get-window)
+        bg-opacity                        (reanimated/use-shared-value 0)
+        translate-y                       (reanimated/use-shared-value window-height)
+        sheet-gesture                     (get-sheet-gesture translate-y
+                                                             bg-opacity
+                                                             window-height
+                                                             on-close)
+        selected-item-smaller-than-sheet? (< @item-height
+                                             (- window-height
+                                                @sheet-height
+                                                (:top insets)
+                                                (:bottom insets)
+                                                bottom-margin))
+        top                               (- window-height (:top insets) @sheet-height)
+        bottom                            (if selected-item-smaller-than-sheet?
+                                            (+ @sheet-height bottom-margin)
+                                            (:bottom insets))]
+    (rn/use-effect
+     #(if hide?
+        (hide translate-y bg-opacity window-height on-close)
+        (show translate-y bg-opacity))
+     [hide?])
```

Comment:
Would this be a reason to remove the use-effect deprecation, or would this be refactored away from use-effect?

Reply:
good question, i need to take a look